eth-rpc/server.nim

```diff
@@ -9,10 +12,8 @@ type
   # Procedure signature accepted as an RPC call by server
   RpcProc* = proc (params: JsonNode): Future[JsonNode]
 
-  RpcServer* = ref object
-    socket*: AsyncSocket
-    port*: Port
-    address*: string
+  RpcServer* = ref object of RootRef
```

Comment:
Is there a reason for `RootRef`? By using `RootRef` `RpcServer` essentially becomes inheritable.

Reply:
But what the reason not to make it inherited? Maybe some other developers wants to store more data in `RpcServer`, e.g. some state data.

Reply:
Ok, fair enough

Reply:
Actually not that fair, because there's no way for the user to initialize his subclass. The only initializer we have is `newRpcServer: RpcServer`

Reply:
Agree, this is one more TODO to have more optional init procedure.